gene_benchmark/descriptor.py

```diff
@@ -4,6 +4,57 @@
 
 import mygene
 import pandas as pd
+import requests
+
+
+def _gene_symbol_to_ensemble_ids(
+    symbols: list[str], species: str = "human"
+) -> dict[str, str]:
+    """
+    converts between symbols to their ensemble ids.
+
+    Args:
+    ----
+        symbols (list[str]): A list of gene symbols
+        species (str, optional): The species for the ensemble id conversion. Defaults to "human".
+
+    Returns:
+    -------
+        dict[str, str]: dictionary with symbol ensemble pairs (if symbol didn't find a ensemble id it will not appear)
+
+    """
+    mg = mygene.MyGeneInfo()
+    query = " ".join(symbols)
+    gene_info = mg.query(query, fields="symbol,ensembl.gene", species=species)
+    ids = {}
+    for hit in gene_info["hits"]:
+        symbol = hit["symbol"]
+        ensembl_id = hit.get("ensembl", {}).get("gene", None)
+        ids[symbol] = ensembl_id
+    return ids
+
+
+def _fetch_ensembl_sequence(ensembl_gene_id):
+    """
+    retries the base pair sequence of a given ensemble id using ensembl.org API.
+
+    Args:
+    ----
+        ensembl_gene_id (str): an ensemble id
+
+    Returns:
+    -------
+        str: base pair sequence of the gene
+
+    """
+    if not ensembl_gene_id:
+        return None
+    url = f"https://rest.ensembl.org/sequence/id/{ensembl_gene_id}?content-type=text/plain"
+    response = requests.get(url)
+    if response.status_code == 200:
```

Comment:
are you sure it will always be 200? 
maybe we could check if it returned an empty string? that way its also more readable and also less prone to break.

Reply:
This is the convention. Maybe because it is API specific it's fine
 https://umbraco.com/knowledge-base/http-status-codes/#:~:text=The%20200%20OK%20status%20code,transmitted%20to%20the%20message%20body.